ostree-ng.sh

```diff
@@ -485,13 +489,129 @@ greenprint "📥 Downloading the image"
+poweroff
+%post --log=/var/log/anaconda/post-install.log --erroronfail
+# no sudo password for user admin and httpbootuser
+echo -e 'admin\tALL=(ALL)\tNOPASSWD: ALL' >> /etc/sudoers
+echo -e 'httpbootuser\tALL=(ALL)\tNOPASSWD: ALL' >> /etc/sudoers
+# add remote prod edge repo
+ostree remote delete rhel
+ostree remote add --no-gpg-verify --no-sign-verify rhel ${PROD_REPO_URL}
+%end
+STOPHERE
+
+# Update grub.cfg to work with HTTP boot
+greenprint "📝 Update grub.cfg to work with HTTP boot"
+sudo tee -a "${GRUB_CFG}" > /dev/null << EOF
+menuentry 'Install Red Hat Enterprise Linux for Edge' --class fedora --class gnu-linux --class gnu --class os {
+        linuxefi /httpboot/images/pxeboot/vmlinuz inst.stage2=http://192.168.100.1/httpboot inst.ks=http://192.168.100.1/ks.cfg inst.text console=ttyS0,115200
+        initrdefi /httpboot/images/pxeboot/initrd.img
+}
+EOF
+sudo sed -i 's/default="1"/default="3"/' "${GRUB_CFG}"
+sudo sed -i 's/timeout=60/timeout=10/' "${GRUB_CFG}"
+
+greenprint "📋 Create libvirt image disk"
+LIBVIRT_HTTP_IMAGE_PATH=/var/lib/libvirt/images/${IMAGE_KEY}-httpboot.qcow2
+sudo qemu-img create -f qcow2 "${LIBVIRT_HTTP_IMAGE_PATH}" 20G
+
+greenprint "📋 Install edge vm via http boot"
+sudo virt-install --name="${IMAGE_KEY}-httpboot"\
+                  --disk path="${LIBVIRT_HTTP_IMAGE_PATH}",format=qcow2 \
+                  --ram 3072 \
+                  --vcpus 2 \
+                  --network network=integration,mac=34:49:22:B0:83:32 \
+                  --os-type linux \
+                  --os-variant "$OS_VARIANT" \
+                  --pxe \
+                  --boot uefi,loader_ro=yes,loader_type=pflash,nvram_template=/usr/share/edk2/ovmf/OVMF_VARS.fd,loader_secure=no \
+                  --nographics \
+                  --noautoconsole \
+                  --wait=-1 \
+                  --noreboot
+
+# Start VM.
+greenprint "💻 Start HTTP BOOT VM"
+sudo virsh start "${IMAGE_KEY}-httpboot"
+
+# Check for ssh ready to go.
+greenprint "🛃 Checking for SSH is ready to go"
+for LOOP_COUNTER in $(seq 0 30); do
+    RESULTS="$(wait_for_ssh_up $HTTP_GUEST_ADDRESS)"
+    if [[ $RESULTS == 1 ]]; then
+        echo "SSH is ready now! 🥳"
+        break
+    fi
+    sleep 10
+done
+
+# Check image installation result
+check_result
+
+# Get ostree commit value.
+greenprint "🕹 Get ostree install commit value"
+INSTALL_HASH=$(curl "${PROD_REPO_URL}/refs/heads/${OSTREE_REF}")
+
+# Add instance IP address into /etc/ansible/hosts
+sudo tee "${TEMPDIR}"/inventory > /dev/null << EOF
+[ostree_guest]
+${HTTP_GUEST_ADDRESS}
+
+[ostree_guest:vars]
+ansible_python_interpreter=/usr/bin/python3
+ansible_user=admin
+ansible_private_key_file=${SSH_KEY}
+ansible_ssh_common_args="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+ansible_become=yes 
```

Comment:
Please remove line 599 and 600, because user admin does not need sudo password (You configured this on kickstart file already). Thanks.